src/Inventory/Asset/Unmanaged.php

```diff
@@ -188,18 +165,25 @@ public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = [])
             //if found, the Unmanaged device has been converted
             $need_to_add = false;
 
-            if (property_exists($val, "mac")) {
-                $result = NetworkPortInstantiation::getUniqueItemByMac(
-                    $val->mac,
-                    $entities_id
-                );
+            if (property_exists($val, "remote_addr")) {
+                $converted_itemtype = null;
+                $converted_items_id = null;
+                foreach ($CFG_GLPI['inventory_types'] as $inventory_type) {
+                    $asset = new $inventory_type();
+                    if ($asset->getFromDBBycrit(['remote_addr' => $val->remote_addr])) {
```

Comment:
There is no unicity on IP address. Another asset may have been inventoried on the same IP since Unmanaged item creation.